api/src/main/java/org/openmrs/module/m2sysbiometrics/bioplugin/BioServerClient.java

```diff
@@ -1,109 +1,20 @@
-                       String biometricXml) {
-        Register register = new Register();
-        register.setLocationID(locationId);
-        register.setID(subjectId);
+    String identify(M2SysClient client, String biometricXml);
 
-        register.setBiometricXml(biometricXml);
+    String delete(M2SysClient client, String subjectId);
 
-        RegisterResponse response = (RegisterResponse) getWebServiceTemplate()
-                .marshalSendAndReceive(serviceUrl, register);
-
-        return response.getRegisterResult();
-    }
-
-    public String update(String serviceUrl, String subjectId, int locationId,
-                       String biometricXml) {
-        Update update = new Update();
-        update.setLocationID(locationId);
-        update.setID(subjectId);
-
-        update.setBiometricXml(biometricXml);
-
-        UpdateResponse response = (UpdateResponse) getWebServiceTemplate()
-                .marshalSendAndReceive(serviceUrl, update);
-
-        return response.getUpdateResult();
-    }
-
-    public String getInfo(String serviceUrl) {
-        GetInfo getInfo = new GetInfo();
-
-        GetInfoResponse response = (GetInfoResponse) getWebServiceTemplate()
-                .marshalSendAndReceive(serviceUrl, getInfo);
-
-        return response.getGetInfoResult();
-    }
-
-    public String changeId(String serviceUrl, String oldId, String newId) {
-        ChangeID changeID = new ChangeID();
-        changeID.setNewID(newId);
-        changeID.setOldID(oldId);
-
-        ChangeIDResponse response = (ChangeIDResponse) getWebServiceTemplate()
-                .marshalSendAndReceive(serviceUrl, changeID);
-
-        return response.getChangeIDResult();
-    }
-
-    public String identify(String serviceUrl, int locationId,
-                         String biometricXml) {
-        Identify identify = new Identify();
-
-        identify.setBiometricXml(biometricXml);
-        identify.setLocationID(locationId);
-
-        IdentifyResponse response = (IdentifyResponse) getWebServiceTemplate()
-                .marshalSendAndReceive(serviceUrl, identify);
-
-        return response.identifyResult;
-    }
-
-    public String delete(String serviceUrl, String subjectId) {
-        DeleteID deleteID = new DeleteID();
-        deleteID.setID(subjectId);
-
-        DeleteIDResponse response = (DeleteIDResponse) getWebServiceTemplate()
-                .marshalSendAndReceive(serviceUrl, deleteID);
-
-        return response.getDeleteIDResult();
-    }
-
-    public String isRegistered(String serviceUrl, String subjectId) {
-        IsRegistered isRegistered = new IsRegistered();
-        isRegistered.setID(subjectId);
-
-        IsRegisteredResponse response = (IsRegisteredResponse) getWebServiceTemplate()
-                .marshalSendAndReceive(serviceUrl, isRegistered);
-
-        return response.getIsRegisteredResult();
-    }
+    String getServiceUrl(M2SysClient client);
```

Comment:
This seems more like an internal method - it belongs more in the abstract class, not the interface

Reply:
Moreover the bioplugin server should not use the client, the client should use the server. If the implementations need the property, they should get it from the admin service.

The get property mehod is something that should be part of the client interface it should stay private.